### PR TITLE
doc/glossary: add "mds" term

### PR DESCRIPTION
**Pull request by zdover23**
Improve "MDS" glossary term. Link to MDS information.

Signed-off-by: Zac Dover <zac.dover@gmail.com>

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [x] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: doc/glossary.rst [hunk 1]
```diff
@@ -184,8 +186,12 @@
...
                 relationship with OSDs.
 
-	MDS
-		The Ceph metadata software.
+	:ref:`MDS<cephfs_add_remote_mds>`
+                The Ceph metadata server daemon. Also referred to as
+                "ceph-mds". The Ceph metadata server daemon is required to run
+                the CephFS file system. The MDS (the daemon) stores all
+                filesystem metadata in the memory of high-availability Ceph
+                Metadata Servers.
```
Comment (anthonyeleven) [inline doc/glossary.rst:194]: This seems self-referential.  

s/(the daemon)//
Suggest truncating after “filesystem metadata”

**Review by anthonyeleven — APPROVED** [review]
Non-blocking suggestions left, I think this wording is awkward.

File: doc/glossary.rst (hunk 1 above)
Comment (zdover23) [reply doc/glossary.rst:194]: I see your point. I'm going to make that alteration.

**Comment by anthonyeleven** [comment]
Lgtm

**Comment by zdover23** [comment]
https://github.com/ceph/ceph/pull/48871 - Quincy backport
https://github.com/ceph/ceph/pull/48872 - Pacific backport

**Pull request merged**